src/mscorlib/shared/System/Reflection/ReflectionTypeLoadException.cs

```diff
@@ -24,9 +26,17 @@ public ReflectionTypeLoadException(Type[] classes, Exception[] exceptions, strin
             HResult = HResults.COR_E_REFLECTIONTYPELOAD;
         }
 
+        private ReflectionTypeLoadException(SerializationInfo info, StreamingContext context) 
+            : base(info, context)
+        {
+            LoaderExceptions = (Exception[])(info.GetValue("Exceptions", typeof(Exception[])));
+        }
+
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             base.GetObjectData(info, context);
+            info.AddValue("Types", null, typeof(Type[]));
```

Comment:
What happens if you do this given that Type isn't serializable?

Reply:
in this case it either uses no FullyQualifiedAssemblyName as you pass null into it or uses the FQAN from System.Array which is serializable. I believe I could also simply remove the typeof(Type[]) but it's not necessary (just checked with the tests).

But to be noted, serialization from Desktop to Core won't either work here...